gems/sorbet-runtime/lib/types/private/final.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+# typed: false
+
+module T::Private::Final
+  module NoInherit
+    def inherited(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be inherited"
+    end
+  end
+
+  module NoIncludeExtend
+    def included(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be included"
+    end
+
+    def extended(arg)
+      super(arg)
+      raise "#{self.name} was declared as final and cannot be extended"
+    end
+  end
+
+  def self.declare(mod)
+    if !mod.is_a?(Module)
+      raise "#{mod.name} is not a class, module or method and cannot be declared as final"
```

Comment:
Given that declaring methods final is done somehow else, does it make sense to even mention `method` here? I can imagine it might just confuse people. Maybe say "declared as final with `final!`"?